src/Testcontainers.K3s/K3sContainer.cs

```diff
@@ -0,0 +1,32 @@
+namespace Testcontainers.K3s;
+
+/// <inheritdoc cref="DockerContainer" />
+[PublicAPI]
+public sealed class K3sContainer : DockerContainer
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="K3sContainer" /> class.
+    /// </summary>
+    /// <param name="configuration">The container configuration.</param>
+    /// <param name="logger">The logger.</param>
+    public K3sContainer(K3sConfiguration configuration, ILogger logger)
+        : base(configuration, logger)
+    {
+    }
+
+    /// <summary>
+    /// Gets the Kubeconfig.
+    /// </summary>
+    /// <returns>Task that completes when the Kubeconfig has been read.</returns>
+    public async Task<string> GetKubeconfigAsync()
+    {
+        var kubeconfigBytes = await ReadFileAsync("/etc/rancher/k3s/k3s.yaml")
+            .ConfigureAwait(false);
+
+        var kubeconfig = Encoding.Default.GetString(kubeconfigBytes);
+
+        var server = new UriBuilder(Uri.UriSchemeHttps, Hostname, GetMappedPublicPort(K3sBuilder.KubeSecurePort)).ToString();
+
+        return Regex.Replace(kubeconfig, "server:\\s?[:/\\.\\d\\w]+", "server: " + server, RegexOptions.None, TimeSpan.FromSeconds(1));
+    }
```

Comment:
@adar2 I changed getting the Kubeconfig file to the code above. Please let me know what do you think (I adjusted the test too). I would like to avoid blocking the sync context and remove the 3rd party dependency. Although the intention is nice, we try not to rely on 3rd party dependencies yet to avoid dependency clashes. I think modifying that small part works without YamlDotNet. If you are fine with the latests changes I am happy to merge the PR.